src/interactions/interaction.ts

```diff
@@ -44,7 +44,9 @@ module Plottable {
      * @return {Interaction}
      */
     public detachFrom(component: Component) {
-      this._unanchor();
+      if (this._isAnchored) {
```

Comment:
Not really because we want detachFrom to revert the interaction to its original state (not be anchored to anything). Say this bit of code
```typescript
var interaction = new Plottable.Interaction();
var component = new Plottable.Component();
interaction.attachTo(component); // nothing anchored yet
interaction.detachFrom(component);

assert(interaction._componentAttachedTo === null)